modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -2489,6 +2670,357 @@ private void _sendPaymentStatusMessage(
 			});
 	}
 
+	private void _setAccountDefaultBillingAddress(
+			CommerceChannel commerceChannel, CommerceOrder commerceOrder)
+		throws PortalException {
+
+		if (commerceOrder.getBillingAddressId() > 0) {
+			return;
+		}
+
+		CommerceChannelAccountEntryRel commerceChannelAccountEntryRel =
+			_commerceChannelAccountEntryRelLocalService.
+				fetchCommerceChannelAccountEntryRel(
+					commerceOrder.getCommerceAccountId(),
+					commerceChannel.getCommerceChannelId(),
+					CommerceChannelAccountEntryRelConstants.
+						TYPE_BILLING_ADDRESS);
+
+		if (commerceChannelAccountEntryRel == null) {
+			return;
+		}
+
+		CommerceAddress commerceAddress =
+			_commerceAddressLocalService.getCommerceAddress(
+				commerceChannelAccountEntryRel.getClassPK());
+
+		if (commerceAddress == null) {
+			return;
+		}
+
+		List<CommerceAddress> billingCommerceAddresses =
+			_commerceAddressLocalService.getBillingCommerceAddresses(
+				commerceChannel.getCommerceChannelId(),
+				AccountEntry.class.getName(),
+				commerceOrder.getCommerceAccountId(), QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS);
+
+		if (billingCommerceAddresses.contains(commerceAddress)) {
+			commerceOrder.setBillingAddressId(
+				commerceAddress.getCommerceAddressId());
+		}
+	}
+
+	private void _setAccountDefaultDeliveryTerm(
+		CommerceChannel commerceChannel, CommerceOrder commerceOrder,
+		User user) {
+
+		if (commerceOrder.getDeliveryCommerceTermEntryId() > 0) {
+			return;
+		}
+
+		CommerceChannelAccountEntryRel commerceChannelAccountEntryRel =
+			_commerceChannelAccountEntryRelLocalService.
+				fetchCommerceChannelAccountEntryRel(
+					commerceOrder.getCommerceAccountId(),
+					commerceChannel.getCommerceChannelId(),
+					CommerceChannelAccountEntryRelConstants.TYPE_DELIVERY_TERM);
+
+		if ((commerceChannelAccountEntryRel == null) ||
+			!commerceChannelAccountEntryRel.isOverrideEligibility()) {
+
+			return;
+		}
+
+		CommerceShippingFixedOption commerceShippingFixedOption =
+			_commerceShippingFixedOptionLocalService.
+				fetchCommerceShippingFixedOption(
+					commerceOrder.getCompanyId(),
+					commerceOrder.getShippingOptionName());
```

Comment:
Another observation here. I see that we introduced this new change in this PR, which was not previously included in the PR sent to BChan previously. Is this an edge case scenario you identified? This would be another clarification request along with my earlier comment regarding removal of `_setAccountDefaultDeliveryTerm` from `addCommerceOrder`